src/Synology.Api.Client.Tests/FileStationApiUploadEndpointTests.cs

```diff
@@ -84,24 +239,70 @@ public void UploadEndpoint_UploadFilePathNotFound_DisplaysCorrectError()
                 }
             };
 
-            _httpTest.RespondWithJson(expectedResponse);
+            //expect certain request to server
+            var request = _mockHtttp.Expect(System.Net.Http.HttpMethod.Post, _baseUriWithApiPath.ToString())
+                 .WithQueryString(new Dictionary<string, string>
+                 {
+                    { "_sid" , _session.Sid }
+                 });
+
+            request.Respond(HttpStatusCode.OK, JsonContent.Create(expectedResponse));
+
+            var fileStationUploadEndpoint = GetFileStationUploadEndpoint();
 
-            // act
-            Func<Task> action = async () => await _fileStationUploadEndpoint.UploadAsync(TestFilePath, destination, true);
+            // Act
+            var actDelegate = () => fileStationUploadEndpoint.UploadAsync(FILEPATH_TO_UPLOAD, _destination, overwrite: true);
 
-            // assert
-            action
-                .Should()
-                .ThrowAsync<SynologyApiException>().Result
-                .And
-                .ErrorDescription
-                .Should()
-                .BeEquivalentTo(ErrorMessages.FileStationApiErrors.GetValueOrDefault(notFoundErrorCode));
+            // Assert
+            var apiException = await Assert.ThrowsAsync<SynologyApiException>(actDelegate);
+            apiException.ErrorDescription.Should().BeEquivalentTo(expectedErrorMessage);
         }
 
-        public void Dispose()
+        [Theory]
+        [InlineData(2, true, "true")]
+        [InlineData(2, false, "false")]
+        [InlineData(3, true, "overwrite")]
+        [InlineData(3, false, "skip")]
+        public async Task Upload_DifferentApiVersion_ShouldSendCorrectValuesForOverwriteAndVersion(int apiVersion, bool overwrite, string overwriteValue)
         {
-            _httpTest.Dispose();
+            // Arrange
+            var apiInfo = new ApiInfo(_apiInfo.Name, _apiInfo.Path, apiVersion, _apiInfo.SessionName);
+
+            var expectedRequestContentApiVersion = $"Content-Disposition: form-data; name=\"version\"{Environment.NewLine}{Environment.NewLine}{apiVersion}";
```

Comment:
This test is failing for me on macOS because it's using `\n` for new lines here. I was looking at the source code for `MultipartContent` and it looks like it always uses `\r\n` so it should be safe to hard code it here in the test.